File: sdk/javascript-tunzisha/TunzishaAPIClient.js
const axios = require('axios');

class HealthcareAPIClient {
    constructor(baseURL) {
        this.baseURL = baseURL;
    }

    async addPatient(patientData) {
        try {
            const response = await axios.post(`${this.baseURL}/api/Patients`, patientData);
            return response.data;
        } catch (error) {
            console.error('Error adding patient:', error.response.data);
            throw error;
        }
    }

    async getAllPatients() {
        try {
            const response = await axios.get(`${this.baseURL}/api/Patients`);
            return response.data;
        } catch (error) {
            console.error('Error getting all patients:', error.response.data);
            throw error;
        }
    }

    async getPatientById(patientId) {
        try {
            const response = await axios.get(`${this.baseURL}/api/Patients/${patientId}`);
            return response.data;
        } catch (error) {
            console.error('Error getting patient by ID:', error.response.data);
            throw error;
        }
    }

    async deletePatient(patientId) {
        try {
            const response = await axios.delete(`${this.baseURL}/api/Patients/${patientId}`);
            return response.data;
        } catch (error) {
            console.error('Error deleting patient:', error.response.data);
            throw error;
        }
    }

    async updatePatient(patientId, updatedPatientData) {
        try {
            const response = await axios.put(`${this.baseURL}/api/Patients/${patientId}`, updatedPatientData);
            return response.data;
        } catch (error) {
            console.error('Error updating patient:', error.response.data);
            throw error;
        }
    }
}

// Example usage:
const apiClient = new HealthcareAPIClient('https://localhost:3000');

const patientData = {
    patientId: 125,
    firstName: 'Dennis',
    secondName: 'Otieno',
    age: 25,
    gender: 'Male',
    profileImage: 'YAYGSLSMNJAMAN',
    contactNumber: '+254743992235',
    nextOfKin: 'Michael Omolo',
    dateOfBirth: '01/02/1999',
    inpatient: true,
    symptoms: 'breathlessness coughing dry_throat',
    diagnosis: 'Covid-19 Omicron',
    prescription: 'Vitamin C and Zinc tablets, each 1 x 3'
};

async function test() {
    try {
        console.log('Add Patient Response:', await apiClient.addPatient(patientData));
        console.log('Get All Patients Response:', await apiClient.getAllPatients());
        const patientId = '65ee32e5390498ef1815c685';
        console.log('Get Single Patient Response:', await apiClient.getPatientById(patientId));
        console.log('Delete Patient Response:', await apiClient.deletePatient(patientId));
        const updatedPatientData = {
            patientId: '65ee32e5390498ef1815c685',
            firstName: 'Updated Name',
            age: 30
        };
        console.log('Update Patient Response:', await apiClient.updatePatient(patientId, updatedPatientData));
    } catch (error) {
        console.error('Error:', error);
    }
}

test();

File: sdk/python-tunzisha/TunzishaAPIClient.py
import requests
import json

class HealthcareAPIClient:
    def __init__(self, base_url):
        self.base_url = base_url

    def add_patient(self, patient_data):
        url = f"{self.base_url}/api/Patients"
        response = requests.post(url, json=patient_data)
        return response.json()

    def get_all_patients(self):
        url = f"{self.base_url}/api/Patients"
        response = requests.get(url)
        return response.json()

    def get_patient_by_id(self, patient_id):
        url = f"{self.base_url}/api/Patients/{patient_id}"
        response = requests.get(url)
        return response.json()

    def delete_patient(self, patient_id):
        url = f"{self.base_url}/api/Patients/{patient_id}"
        response = requests.delete(url)
        return response.json()

    def update_patient(self, patient_id, updated_patient_data):
        url = f"{self.base_url}/api/Patients/{patient_id}"
        response = requests.put(url, json=updated_patient_data)
        return response.json()

# Example usage:
api_client = HealthcareAPIClient("https://localhost:3000")
patient_data = {
    "patientId": 125,
    "firstName": "Dennis",
    "secondName": "Otieno",
    "age": 25,
    "gender": "Male",
    "profileImage": "YAYGSLSMNJAMAN",
    "contactNumber": "+254743992235",
    "nextOfKin": "Michael Omolo",
    "dateOfBirth": "01/02/1999",
    "inpatient": True,
    "symptoms": "breathlessness coughing dry_throat",
    "diagnosis": "Covid-19 Omicron",
    "prescription": "Vitamin C and Zinc tablets, each 1 x 3"
}
print("Add Patient Response:", api_client.add_patient(patient_data))

print("Get All Patients Response:", api_client.get_all_patients())

patient_id = "65ee32e5390498ef1815c685"
print("Get Single Patient Response:", api_client.get_patient_by_id(patient_id))

print("Delete Patient Response:", api_client.delete_patient(patient_id))

updated_patient_data = {
    "patientId": "65ee32e5390498ef1815c685",
    "firstName": "Updated Name",
    "age": 30
}
print("Update Patient Response:", api_client.update_patient(patient_id, updated_patient_data))
